ui/app/src/components/ComponentMenu.tsx

```diff
@@ -0,0 +1,235 @@
+    <>
+      <Menu
+        anchorEl={anchorEl}
+        open={Boolean(anchorEl)}
+        onClose={handleClose}
+        anchorOrigin={anchorOrigin}
+      >
+        <MenuItem onClick={() => handleEdit('form')}>
+          <FormattedMessage
+            id="previewToolBar.menu.edit"
+            defaultMessage="Edit"
+          />
+        </MenuItem>
+        {
+          (!publishDialog.item?.lockOwner && !publishDialog.item?.isLive) &&
+          <MenuItem onClick={() => handleEdit('schedule')}>
+            <FormattedMessage
+              id="previewToolBar.menu.schedule"
+              defaultMessage="Schedule"
+            />
+          </MenuItem>
+        }
+        {
+          (!publishDialog.item?.lockOwner && !publishDialog.item?.isLive) &&
+          <MenuItem onClick={() => handleEdit('publish')}>
+            <FormattedMessage
+              id="previewToolBar.menu.publish"
+              defaultMessage="Publish"
+            />
+          </MenuItem>
+        }
+        <MenuItem onClick={handleClose}>
+          <FormattedMessage
+            id="previewToolBar.menu.history"
+            defaultMessage="History"
+          />
+        </MenuItem>
+        <MenuItem onClick={handleClose}>
+          <FormattedMessage
+            id="previewToolBar.menu.dependencies"
+            defaultMessage="Dependencies"
+          />
+        </MenuItem>
+        <MenuItem onClick={handleClose}>
+          <FormattedMessage
+            id="previewToolBar.menu.delete"
+            defaultMessage="Delete"
+          />
+        </MenuItem>
+        <MenuItem onClick={handleClose} className={classes.separator}>
+          <FormattedMessage
+            id="previewToolBar.menu.infoSheet"
+            defaultMessage="Info Sheet"
+          />
+        </MenuItem>
+        <MenuItem onClick={() => handleEdit('template')}>
+          <FormattedMessage
+            id="previewToolBar.menu.editTemplate"
+            defaultMessage="Edit Template"
+          />
+        </MenuItem>
+        {
+          publishDialog.item && !embeddedParentPath && contentTypesBranch.byId[publishDialog.item.contentType]?.type === 'page' &&
+          <MenuItem onClick={() => handleEdit('controller')}>
+            <FormattedMessage
+              id="previewToolBar.menu.editController"
+              defaultMessage="Edit Controller"
+            />
+          </MenuItem>
+        }
+      </Menu>
+      {
+        dialogConfig.open &&
+        <EmbeddedLegacyEditors
+          dialogConfig={dialogConfig}
+          setDialogConfig={setDialogConfig}
+          getPath={getPath}
+          showController={!embeddedParentPath && contentTypesBranch.byId[publishDialog.item.contentType]?.type === 'page'}
+        />
+      }
+      {
+        publishDialog.open &&
+        <PublishDialog
+          scheduling={publishDialog.scheduling}
+          items={[publishDialog.item]}
+          onSuccess={onSuccessPublish}
+          onClose={onClosePublish}
+        />
```

Comment:
We could set these higher above in the component tree so less components would require this code within them.

Reply:
pending